omod/src/test/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResourceTest.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * The contents of this file are subject to the OpenMRS Public License
+ * Version 1.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * http://license.openmrs.org
+ *
+ * Software distributed under the License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+ * License for the specific language governing rights and limitations
+ * under the License.
+ *
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+package org.openmrs.module.idgen.rest.resource;
+
+
+import org.openmrs.module.idgen.IdentifierSource;
+import org.openmrs.module.idgen.service.IdentifierSourceService;
+import org.junit.Before;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResourceTest;
+
+
+public class IdentifierSourceResourceTest extends BaseDelegatingResourceTest<IdentifierSourceResource, IdentifierSource> {
+
+	public final static String IDENTIFIER_SOURCE_UUID = "0d47284f-9e9b-4a81-a88b-8bb42bc0a999";
+	
+	@Before
+	public void setup() throws Exception {
```

Comment:
What is this one for? Is this dataset used by any methods in this class?

Reply:
Yes. I am using the dataset in the `newObject()` method.

Reply:
In the example provided in the ticket, the newObject method returns: `Context.getConceptService().getConceptMapTypeByUuid(getUuidProperty());`

the getUuidProperty() method `RestTestConstants1_9.CONCEPT_MAP_TYPE_UUID; `

It seems that there is some dataset being used.

Reply:
Just create a new object without the dataset.

Reply:
please comment on my approach:

`SequentialIdentifierGenerator identifierGenerator = new SequentialIdentifierGenerator();
identifierGenerator.setUuid(UUID);
Context.getService(IdentifierSourceService.class).saveIdentifierSource(identifierGenerator);
return identifierGenerator;`

Reply:
Why not just do new SequentialIdentifierGenerator()

Reply:
getting this error: Tests in error: 
  `asRepresentation_shouldReturnValidRefRepresentation(org.openmrs.module.idgen.rest.resource.IdentifierSourceResourceTest): Don't know how to get IdentifierSourceResource(class org.openmrs.module.idgen.SequentialIdentifierGenerator) as ref`

Reply:
Is that caused by new SequentialIdentifierGenerator()?
If yes, can you paste the stack trace at pastebin.com for me to take a look?

Reply:
https://pastebin.com/HYm6XFBy

Reply:
Just commit all your changes. I will try them out locally to find the cause of the exception.